### ISSUE TITLE
bug: Status bar coloring issues on theme patch

### ISSUE THREAD

**Issue by yashslakhtariya** [ReVanced/revanced-patches#4001]
### Bug description

![Image](https://github.com/user-attachments/assets/7152c1e8-a98d-4368-851f-70ef83a4381b)

When custom theme or any predefined theme including Material You in either light/dark mode is applied, status bar coloring issue is there, please find attached screenshot.

Using latest patches 5.1.0 for latest suggested version 19.45.38, used patches are attached.

[selected_patches_2024-11-27_09_00_31.json](https://github.com/user-attachments/files/17928462/selected_patches_2024-11-27_09_00_31.json)

### Error logs

```shell

```

### Acknowledgements

- [x] I have checked all open and closed bug reports and this is not a duplicate.
- [x] I have chosen an appropriate title.
- [x] All requested information has been provided properly.

**Comment by oSumAtrIX** [comment]
It is a transparent overlay of black iirc

**Comment by yashslakhtariya** [comment]
Yes the status bar transparency is the problem. For light themes, it is white like and for dark theme, black like.

Also the text is white on some shades of light theme while black on some shades. Tried given option 'Light Red' for light theme and 'Catppuccin Mocha' for dark theme.

Also happens on latest patches 5.2.0 and YT 19.46.42

**Comment by oSumAtrIX** [comment]
Pretty sure its just another color shade that needs to be changed by the theme patch. Maybe its also some Android specific thing regarding status bar customization.

**Comment by yashslakhtariya** [comment]
I tried so, also tried material you options, not solving the issue

**Comment by yashslakhtariya** [comment]
> @yashslakhtariya what custom colors are you using?  The patch selection JSON file does no include patch options.

Light : #FFFFD8DB
Dark : Catppuccin Mocha (#FF181825)

**Comment by yashslakhtariya** [comment]
The option of transparent status bar isn't there. And also using different color isn't the permanent solution.

If possible devs can give option to theme status bar also with the theme patch color provided in options. Or the option to atleast make it transparent and selectable foreground colors (white/black)

I know it is hard to implement but I just hope devs can do something. They have done so many things in a perfect way.
